src/dodal/devices/attenuator/attenuator.py

```diff
@@ -83,3 +86,29 @@ async def set(self, value: float):
                 for i in range(16)
             ]
         )
+
+
+class EnumFilterAttenuator(ReadOnlyAttenuator):
+    """The attenuator will insert filters into the beam to reduce its transmission.
+
+    This device is currently working, but feature incomplete. See https://github.com/DiamondLightSource/dodal/issues/972
+
+    In this attenuator, the state of a filter corresponds to the selected material,
+    e.g Ag50, in contrast to either being 'IN' or 'OUT'; see BinaryFilterAttenuator.
+    """
+
+    def __init__(
+        self,
+        num_filters: int,
+        filter_selection: list[type[FilterSelection]],
+        prefix: str,
+        name: str = "",
+    ):
+        with self.add_children_as_readables():
+            self.filters: DeviceVector[FilterMotor] = DeviceVector(
+                {
+                    i: FilterMotor(filter_selection[i], f"{prefix}MP{i+1}:", name)
+                    for i in range(num_filters)
+                }
+            )
+        super().__init__(name=name, prefix=prefix)
```

Comment:
```suggestion
    def __init__(
        self,
        prefix: str,
        filter_selection: tuple[type[FilterSelection]],
        name: str = "",
    ):
        with self.add_children_as_readables():
            self.filters: DeviceVector[FilterMotor] = DeviceVector(
                {
                    index: FilterMotor(filter , f"{prefix}MP{index+1}:", name)
                    for index, filter in enumerate(filter_selection)
                }
            )
        super().__init__(prefix, name=name)
```

Should: If num_filters ever doesn't match len(filter_selection) [e.g. I am making a new device and aren't careful] it will cause issue, and is extra book-keeping. 
Should: Making the filter_selection immutable means we can pass in a default if it ends up saving time later, and generally is good practice if the contents of the Sequence don't need to change (if they do, we can adjust typing to Sequence and support tuple/list as appropriate)
Nit: device_factory doesn't require args to be in a particular order, but we have a general practice of prefix being the first argument, just for usability I'd stick to it.